docs/HowToUseSmartFS.md

```diff
@@ -0,0 +1,139 @@
+# How to use SmartFS
+
+SmartFS is FileSystem used for user partition in TizenRT.
+
+SmartFS stands for Sector Mapped Allocation for Really Tiny (SMART) flash.
+It is a filesystem that has been designed to work primary with small,
+serial NOR type flash parts that are 1M byte to 16M byte in size
+(though this is not a limitation).
+
+The filesystem operates by segmenting the flash (or flash partition)
+into “logical sectors” of equal size and then managing them (allocating,
+mapping, chaining, releasing, etc.) to build files and directories
+
+This guide explains the steps and details for enabling SmartFS on a new board.
+
+## SmartFS Code Layering
+
+Smart FS consists of two layers built on top of stanadard MTD Driver.
+Below diagram shows the dependant layers part of Smart FS.
+
+```
++----------------------------------------------+
+| SMART FS (Smart FileSystem Layer)            |
++----------------------------------------------+
+| SMART MTD Driver (Smart Block Driver Layer)  |
++----------------------------------------------+
+| MTD Driver  (MTD Block Driver Layer)         |
++----------------------------------------------+
+| FLASH Driver (Flash Driver)                  |
++----------------------------------------------+
+```
+
+The code directly above the MTD driver is the SMART MTD layer.
+This interfaces with the MTD (flash driver) layer and handles low-level
+media operations such as logical sector allocation, freeing and management,
+erase block management, low-level formatting, wear leveling, etc.
+
+On top of the SMART MTD layer is the Smart Filesystem Layer.
+The SmartFS code uses the logical sector services of the SMART MTD layer to
+provide file and directory level management, such as creating new files,
+chaining logical sectors together to create files, creating directories and
+ file / directory search and management routines.
+
+## FLASH Driver
+Flash driver must provide low-level read/write/erase capability for the FLASH HW.
+Example driver is at os/arch/arm/src/s5j/s5j_sflash.c
+
+During bootup flash driver must be initialized similar to below example snippet.
+```
+void arm_boot(void)
+{
+	...
+	s5j_board_initialize();
+	...
+}
+
+void s5j_board_initialize(void)
+{
+#ifdef CONFIG_S5J_SFLASH
+        s5j_sflash_init();
+#endif
+}
+```
+### MTD Driver
+MTD driver must be enabled to access the underlying FLASH driver.
+Example driver is at os/fs/driver/mtd/mtd_progmem.c.
+
+Below configuration must enabled for using mtd driver.
```

Comment:
Could you explain how to enable them through menuconfig?
For example,
```
File Systems -> xxx -> yyy to set y
```

Reply:
Okay, I will explain